core/runtime/src/main/java/io/quarkus/runtime/ApplicationConfig.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.runtime;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class ApplicationConfig {
+
+    /**
+     * The name of the application.
+     * If not set, defaults to the name of the project.
+     */
+    @ConfigItem(defaultValue = "${project.name:__UNSET__}")
+    public String name;
+
+    /**
+     * The version of the application.
+     * If not set, defaults to the version of the project
+     */
+    @ConfigItem(defaultValue = "${project.version:__UNSET__}")
+    public String version;
+
+    public static boolean isNameUnset(String name) {
+        return name.equals("__UNSET__");
```

Comment:
```suggestion
        return (name == null) || ("__UNSET__".equals(name));
```

Reply:
For the cases where the Config object is uninitialized properly

Reply:
I'm going to change this stuff based on @dmlloyd's comments, but thanks for the suggestion